src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1121,18 +1121,27 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             int typeID = svThisPtr.metaData.type;
 
             //Test for exact match
-            List<FunctionEndPoint> exactFeps = new List<FunctionEndPoint>();
-
-            foreach (FunctionEndPoint fep in feps)
-                if (fep.ClassOwnerIndex == typeID)
-                    exactFeps.Add(fep);
-
-            if (exactFeps.Count == 1)
+            IEnumerable<FunctionEndPoint> exactFeps = null;
+            // Is static method call (i.e no this pointer)
+            if (svThisPtr.Pointer == Constants.kInvalidIndex)
```

Comment:
Since we already had logic here to handle function endpoint matching based on the this pointer it seems reasonable to add handle static function calls as well (and match them against the class id)

Reply:
how was this matched previously?

Reply:
In the case of a class instance, the function endpoint was matched to the function declared in the derived class (if no more than one func matched the signature form the derived class)
In the case of static class,  the function endpoint was not matched at all.